src/main/scala/io/iohk/ethereum/vm/Storage.scala

```diff
@@ -17,8 +17,20 @@ object Storage {
   */
 class Storage private(private val underlying: Map[DataWord, DataWord] = Map()) {
 
-  def store(addr: DataWord, value: DataWord): Storage = new Storage(underlying + (addr -> value))
+  /** Stores new value in an underlying hashmap.
+    * If DataWord.Zero is passed as a value then a corresponding key is removed from the hashmap
+    */
+  def store(addr: DataWord, value: DataWord): Storage = {
+    val newUnderlying: Map[DataWord, DataWord] = value match {
+      case DataWord.Zero => underlying - addr
```

Comment:
Nevermind, I thought it might be using reference comparison but it's using `equals`, so that's good 👍